pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -240,8 +223,16 @@ func (c *Controller) syncOwnedProvisionRequest(ctx context.Context, wl *kueue.Wo
 		attempt := int32(1)
 		shouldCreatePr := false
 		if exists {
-			attempt = getAttempt(ctx, oldPr, wl.Name, checkName)
-			if apimeta.IsStatusConditionTrue(oldPr.Status.Conditions, autoscaling.Failed) {
+			switch {
+			case isCapacityRevoked(oldPr):
+				if workload.IsActive(wl) && !workload.IsFinished(wl) {
+					if err := c.deactivateWorkload(ctx, wl); err != nil {
+						return nil, err
+					}
+					c.record.Eventf(oldPr, corev1.EventTypeNormal, "CapacityRevoked", "Capacity for %v, has been revoked", oldPr.Name)
```

Comment:
```suggestion
					c.record.Eventf(oldPr, corev1.EventTypeNormal, "CapacityRevoked", "Deactivating workload because capacity for %v", oldPr.Name)
```

Thanks for adding the event.

I regret not having added a reason along side the `active` field to explain why Kueue decided to deactivate the Workload.

But maybe it's just worth adding an Evicted condition just before deactivating that could record the reason?

Reply:
Done